### ISSUE TITLE
[L4D2 Beta/Linux] Missing music in main menu

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/Source-1-Games#416]
After downloading and installing the game, I noticed that the music of the main menu is missing.

Console output:
http://nopaste.info/90f2b6bb25.html

**Comment by kisak-valve** [comment]
@sizeofbool commented on 2015-10-05T18:49:08

This bug is still unresolved. There is a message in game's console:

```
S_StartSound: Failed to load sound '*#music/l4d2/l4d2_c1_pc.mp3', file probably missing from disk/repository
```

This **"*#music"** folder looks corrupted and music file itself is present in **"/left4dead/sound/music/l4d2/l4d2_c1_pc.mp3"**.
Checking game's cache doesn't fix this issue.

**Comment by sizeofbool** [comment]
@kisak-valve there is workaround for this issue mentioned above. As soon as you rename "libmiles.so" to "libMiles.so" in "Left 4 Dead 2/bin" folder, music in main menu will be played.

**Comment by vitacell** [comment]
I still have this bug.

**Comment by TheThunderGuyS** [comment]
Still unfixed following The Last Stand update.

**Comment by BioBox** [comment]
I can testify to this. Still doesn't work.

**Comment by Grammernatzi** [comment]
> @kisak-valve there is workaround for this issue mentioned above. As soon as you rename "libmiles.so" to "libMiles.so" in "Left 4 Dead 2/bin" folder, music in main menu will be played.

Are you serious? It's as simple as a simple renaming of a file and it works? And they still haven't fixed it, five years after you mentioned this, for what would literally take five seconds?

Valve, what the fuck?

**Comment by sizeofbool** [comment]
There is even better workaround to this issue:
You can create symbolic link file, with a name "libMiles.so" linked to real "libmiles.so" library file in "Left 4 Dead 2/bin" folder.
Example:
```
$ cd ~/.steam/steam/steamapps/common/Left\ 4\ Dead\ 2/bin
$ ln -s libmiles.so libMiles.so
```
This way fix won't be reverted if you verify game's cache and also ensure game's library files will be working in case some of them were compiled and linked to "libmiles.so" library file instead of "libMiles.so".

**Comment by kinten108101** [comment]
> @kisak-valve there is workaround for this issue mentioned above. As soon as you rename "libmiles.so" to "libMiles.so" in "Left 4 Dead 2/bin" folder, music in main menu will be played.

Did this but problem persists, even tried making soft link

**Comment by H3CATE** [comment]
> There is even better workaround to this issue: You can create symbolic link file, with a name "libMiles.so" linked to real "libmiles.so" library file in "Left 4 Dead 2/bin" folder. Example:
> 
> ```
> $ cd ~/.steam/steam/steamapps/common/Left\ 4\ Dead\ 2/bin
> $ ln -s libmiles.so libMiles.so
> ```
> 
> This way fix won't be reverted if you verify game's cache and also ensure game's library files will be working in case some of them were compiled and linked to "libmiles.so" library file instead of "libMiles.so".

Still doesn't fix the issue for me on Fedora 36. Also tried changing the speaker type but no luck.

**Comment by Gragt** [comment]
Same here. It seems the files is even named `LibMiles.so` by default now, but the console still displays `S_StartSound: Failed to load sound '*#music/l4d2/l4d2_c1_pc.mp3', file probably missing from disk/repository`.

**Comment by yxnan** [comment]
I tried this and it works for me: https://github.com/ValveSoftware/csgo-osx-linux/issues/395#issuecomment-131240196

TL;DR: Just replace the `libMiles.so` with **Portal 2**'s version, and suddenly everything works.

BTW It's not just the main menu audio is missing, all the custom campaigns with mp3 audio (almost all of them) will not work.

I wrote a guide on how-to: [Steam Community :: Guide](https://steamcommunity.com/sharedfiles/filedetails/?id=3114538222)

**Comment by Gragt** [comment]
By Jove, I can’t believe that worked! Thanks!

**Comment by kisak-valve** [comment]
Hello, per "Fixed music not playing at the main menu on Linux." in the [2024-05-14](https://store.steampowered.com/news/app/550/view/4186736071078420633) Left 4 Dead 2 update, please retest this issue.

**Comment by yxnan** [comment]
@kisak-valve Still not works for me.
I noticed the newly shipped `libmiles.so` is exactly the same file as the old one, which is known to be broken.

```bash
$ ls -al lib{M,m}iles.*
.rwxr-xr-x@ 453k yxnan 17 mei 13:21 libmiles.so
.rwxr-xr-x@ 453k yxnan 29 okt  2022 libMiles.so.orig

$ md5sum lib{M,m}iles.*
cbedd6a484ad0f35674a77a5bf583a7d  libMiles.so.orig
cbedd6a484ad0f35674a77a5bf583a7d  libmiles.so
```

**Comment by stevethepocket** [comment]
Working for me, but this is also my first time launching the game on this machine OR distribution so I couldn't tell you if it was broken before now either.

**Comment by RevengeRip** [comment]
I can confirm that issue **IS NOT** resolved!

**Comment by kisak-valve** [comment]
Hello @OlegAckbar, can you note if your game install predates May 14th and if `libmiles.so` has any capitalized letters? We know that the Steam client has a harder time with changes to filename capitalization (https://github.com/ValveSoftware/steam-for-linux/issues/577). If that's true, then remove libmiles and verify the integrity of the game's files, then retest.

**Comment by RevengeRip** [comment]
@kisak-valve yes, my install indeed was predating May 14th and it had `libMiles.so`. I removed this library and verified integrity of the game's files, now I have `libmiles.so` and main menu music works. I guess issue this issue is resolved.

**Comment by H3CATE** [comment]
I can confirm that after reinstalling the game, deleting the library manually and verifying cache / renaming the library with a capitalized M, switching speaker setup, this issue is still NOT fixed on my end, on Fedora 40. I can hear the UI sound but no background music. Another bug that might be related is that the menu background video is ALWAYS Dead Center and the other background videos are never seen on the linux version. 

Hoping this gets fixed soon :)

**Comment by H3CATE** [comment]
> I can confirm that after reinstalling the game, deleting the library manually and verifying cache / renaming the library with a capitalized M, switching speaker setup, this issue is still NOT fixed on my end, on Fedora 40. I can hear the UI sound but no background music. Another bug that might be related is that the menu background video is ALWAYS Dead Center and the other background videos are never seen on the linux version.
> 
> Hoping this gets fixed soon :)

I can also confirm that replacing the file with the one from Portal 2 doesn't work anymore UNLESS you rename it to lowercase. 

> @kisak-valve Still not works for me. I noticed the newly shipped `libmiles.so` is exactly the same file as the old one, which is known to be broken.
> 
> ```shell
> $ ls -al lib{M,m}iles.*
> .rwxr-xr-x@ 453k yxnan 17 mei 13:21 libmiles.so
> .rwxr-xr-x@ 453k yxnan 29 okt  2022 libMiles.so.orig
> 
> $ md5sum lib{M,m}iles.*
> cbedd6a484ad0f35674a77a5bf583a7d  libMiles.so.orig
> cbedd6a484ad0f35674a77a5bf583a7d  libmiles.so
> ```

It seems the fix that was pushed was to make the game take the right file name instead of switching the libmiles.so version with the one from Portal 2.

**Comment by stevethepocket** [comment]
So am I to understand that the Steam client's patcher isn't capable of replacing a file with one of the same name but with different case? Because if so, we need to make sure that's been filed as a bug somewhere. I remember back when _Dust: an Elysian Tail_ was first ported, it wasn't working for me because I'd tried copying the files over from my Windows drive and it verified despite one or two files being wrong-case, and the game just straight-up wouldn't launch.

**Comment by H3CATE** [comment]
> So am I to understand that the Steam client's patcher isn't capable of replacing a file with one of the same name but with different case? Because if so, we need to make sure that's been filed as a bug somewhere. I remember back when _Dust: an Elysian Tail_ was first ported, it wasn't working for me because I'd tried copying the files over from my Windows drive and it verified despite one or two files being wrong-case, and the game just straight-up wouldn't launch.

My understanding of the issue so far is that the library itself is broken, no matter the name and capitalization and needs to be replaced by a working one. The current fix is to use the one from Portal 2 and rename it to libmiles.so without any capital letter.

**Comment by kisak-valve** [comment]
> Replying to https://github.com/ValveSoftware/Source-1-Games/issues/416#issuecomment-2240054834

Tracked at https://github.com/ValveSoftware/steam-for-linux/issues/577.

**Comment by super23** [comment]
Also having issues with libMiles and mp3 playback.
